### PR TITLE
Add dbt documentation to `sale.*` assets

### PR DESCRIPTION
**Pull request by dfsnow**
This PR adds column documentation to all tables and views in the `sale` schema. It also adds top-level documentation to each table/view. Finally, it correctly capitalizes MyDec elsewhere in our dbt documentation.

@wagnerlmichael please review this for accuracy.

### CODE REVIEW

File: dbt/models/default/schema/default.vw_pin_sale.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ models:
       - name: is_multisale
         description: '{{ doc("shared_column_sale_is_multisale") }}'
       - name: is_mydec_date
-        description: Indicator for whether or not the observation uses the myDec sale date
+        description: Indicator for whether or not the observation uses the MyDec sale date
```
Comment (dfsnow) [inline dbt/models/default/schema/default.vw_pin_sale.yml:17]: Just capitalizing MyDec correctly in all places in the documentation. Didn't want to put it in a separate PR.

File: dbt/models/sale/docs.md [hunk 2]
```diff
@@ -1,19 +1,67 @@
...
 {% docs flag %}
-This table holds the flag information from the sales val program.
+PIN-level sales validation flags created by
+[model-sales-val](https://github.com/ccao-data/model-sales-val).
+
+This is the primary sales validation output table. Flags within this table
+should be possible to reconstruct using the other sales validation tables:
+`sale.group_mean`, `sale.parameter`, and `sale.metadata`.
+
+**Primary Key**: `meta_sale_document_number`, `run_id`, `version`
```
Comment (dfsnow) [inline dbt/models/sale/docs.md:11]: I really like having the docs list the primary key somewhere, but I'm somewhat ambivalent about the format I've used here.

One alternative is to add the primary key as a `meta` item for each model:

```yaml
model:
  - name: test
    meta:
      pk:
        - meta_sale_document_number
        - run_id
        - version
```

It then gets formatted into a kinda ugly string in the **Details** section of the docs, like this:

```
'["meta_sale_document_number", "run_id", "version"]'
```

@jeancochrane @wagnerlmichael What do you all think?

File: dbt/models/sale/docs.md [hunk 3]
```diff
@@ -1,19 +1,67 @@
...
 {% docs parameter %}
-This table holds information about the specifications used to flag outliers in the sales val program.
+Parameters used for each run of
+[model-sales-val](https://github.com/ccao-data/model-sales-val),
+including the statistical bounds, groupings, window sizes, etc.
+
+**Primary Key**: `run_id`
 {% enddocs %}
 
+# group_mean
```
Comment (dfsnow) [inline dbt/models/sale/docs.md:32]: I've added markdown headers to the doc assets just to make them easier to read, navigate, and fold.

File: dbt/models/sale/schema.yml [hunk 4]
```diff
@@ -1,17 +1,154 @@
...
+
+              Sales with prices beyond these boundaries are flagged.
+          - name: earliest_data_ingest
+            description: |
+              Date of earliest sale used in validation.
+
+              This inclusive of the rolling window period used for
+              calculating statistical groups. In other words, if the earliest
+              sale to-be-flagged is 2013-12-01 and the rolling window period
+              is 9 months, then the earliest sale *used* would be 2023-03-01
```
Comment (wagnerlmichael) [inline dbt/models/sale/schema.yml:80]: [**Nitpick**]: Typo here on the year.

File: dbt/models/sale/schema.yml [hunk 5]
```diff
@@ -1,17 +1,154 @@
...
+          - name: iso_forest_cols
+            description: Columns used as features in the isolation forest model
+          - name: latest_data_ingest
+            description: Date of latest sale used in validation
+          - name: min_group_thresh
+            description: |
+              Minimum number of sales required for statistical flagging
```
Comment (wagnerlmichael) [inline dbt/models/sale/schema.yml:87]: [**Suggestion**]: Could be good add more here. Something like - *If the minimum number of sales in our group methodology (township, class, rolling window) is below 30, these sales are not flagged and are set to 'Not outlier'*.

File: dbt/models/sale/schema.yml [hunk 6]
```diff
@@ -1,17 +1,154 @@
...
+          - name: res_stat_groups
+            description: |
+              Groups used to calculate flagging statistics (std. dev.)
+              for residential (class 2) properties
+          - name: rolling_window
+            description: |
+              Rolling window size, in months.
+
+              For each target sale, calculate statistics (std. dev.,
+              group size) using all sales in the period N months prior
```
Comment (wagnerlmichael) [inline dbt/models/sale/schema.yml:101]: [**Nitpick**]: Using all sales within N months prior, *inclusive* of the month of the sale itself.

**Review by wagnerlmichael — COMMENTED** [review]
Looks good! Added some nitpicks and small suggestions.

File: dbt/models/sale/schema.yml (hunk 4 above)
Comment (dfsnow) [reply dbt/models/sale/schema.yml]: Fixed in 4eb71d588cf355607577748ddc6ca63bd96aa298!

File: dbt/models/sale/schema.yml (hunk 5 above)
Comment (dfsnow) [reply dbt/models/sale/schema.yml]: Done in b57e0a154027b40faef2bfe742ef40622cadebb8

**Review by wagnerlmichael — APPROVED** [review]

File: dbt/models/sale/docs.md (hunk 2 above)
Comment (dfsnow) [reply dbt/models/sale/docs.md:11]: Totally, that works for me!

File: dbt/models/sale/docs.md [hunk 7]
```diff
@@ -1,19 +1,67 @@
+# flag
+
 {% docs flag %}
```
[the remark this answers is not in the document]
Comment (dfsnow) [reply dbt/models/sale/docs.md:3]: I've already started to do that in the upcoming #203! So we're definitely on the same page.

File: dbt/models/sale/schema.yml [hunk 8]
```diff
@@ -1,17 +1,159 @@
...
+            description: '{{ doc("shared_column_sv_is_outlier") }}'
+          - name: sv_outlier_type
+            description: '{{ doc("shared_column_sv_outlier_type") }}'
+          - name: version
+            description: '{{ doc("shared_column_sv_version") }}'
+
+      - name: foreclosure
+        description: '{{ doc("foreclosure") }}'
+        tags:
+          - load_manual
```
[the remark this answers is not in the document]
Comment (dfsnow) [reply dbt/models/sale/schema.yml:36]: Probably, but there are a billion columns and I don't want to add docs for them right now. I added this as a subtask to #201.
